design/one-pager-considerations-for-changing-test-framework-of-uptest.md

```diff
@@ -0,0 +1,167 @@
+# Considerations for Changing Test Framework of Uptest
+
+* Owner: Sergen Yalcin (@sergenyalcin)
+* Reviewers: Uptest Maintainers
+* Status: Draft
+
+## Background
+
+Uptest is a tool for testing and validating Crossplane providers. This tool is
+utilized to test the behavior of Crossplane controllers, resource management
+operations, and other Crossplane components. Uptest can simulate the creation,
+update, import, deletion, and other operations of Crossplane resources.
+
+When `Uptest` was first written, it used [kuttl](https://github.com/kudobuilder/kuttl)
+as the underlying test framework in order to have good declarative testing
+capabilities. Over time, it was realized that there are better and more
+compatible alternatives and `kuttl` is a tool that is not currently maintained,
+so it was decided to change the underlying framework.
+
+[kuttl](https://github.com/kudobuilder/kuttl) provides a declarative approach to
+test Kubernetes Operators. `kuttl` is designed for testing operators, however it
+can declaratively test any kubernetes objects.
+
+While starting to the `Uptest` effort, we considered a few different
+alternatives and `kuttl`'s capabilities were appropriate for our assertion
+aims although its missing points. Today, we consider changing the underlying
+test framework tool because of the `kuttl` stopped to be actively maintained.
```

Comment:
```suggestion
When starting the `Uptest` effort, we considered a few different
alternatives and `kuttl`'s capabilities were appropriate for our assertion
aims even though it missed some points. Today, we consider changing the underlying
test framework tool because of the perception of `kuttl` not being actively maintained
and other frameworks offering superior capabilities.
```

Reply:
Done!